vcd/resource_vcd_nsxt_ip_set_test.go

```diff
@@ -288,9 +288,692 @@ const testAccNsxtIpSetDS = `
+					resourceFieldsEqual("data.vcd_nsxt_ip_set.set1", "vcd_nsxt_ip_set.set1", []string{"%"}),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdNsxtIpSetInheritedVdcStep1 = `
+data "vcd_external_network_v2" "existing-extnet" {
+	name = "{{.ExternalNetwork}}"
+}
+
+resource "vcd_nsxt_edgegateway" "nsxt-edge" {
+  org  = "{{.Org}}"
+  name = "{{.NsxtEdgeGatewayVcd}}"
+
+  external_network_id = data.vcd_external_network_v2.existing-extnet.id
+
+  subnet {
+     gateway               = tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].gateway
+     prefix_length         = tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].prefix_length
+
+     primary_ip            = tolist(tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].static_ip_pool)[0].end_address
+     allocated_ips {
+       start_address = tolist(tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].static_ip_pool)[0].end_address
+       end_address   = tolist(tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].static_ip_pool)[0].end_address
+     }
+  }
+}
+
+resource "vcd_nsxt_ip_set" "set1" {
+  org  = "{{.Org}}"
+
+  edge_gateway_id = vcd_nsxt_edgegateway.nsxt-edge.id
+
+  name = "{{.IpSetName}}"
+
+  ip_addresses = [
+    "12.12.12.1",
+    "10.10.10.0/24",
+    "11.11.11.1-11.11.11.2",
+    "2001:db8::/48",
+    "2001:db6:0:0:0:0:0:0-2001:db6:0:ffff:ffff:ffff:ffff:ffff",
+  ]
+}
+`
+
+const testAccVcdNsxtIpSetInheritedVdcStep3 = testAccVcdNsxtIpSetInheritedVdcStep1 + `
+# skip-binary-test: Data Source test
+data "vcd_nsxt_edgegateway" "nsxt-edge" {
+  org  = "{{.Org}}"
+  name = "{{.NsxtEdgeGatewayVcd}}"
+}
+
+data "vcd_nsxt_ip_set" "set1" {
+  org  = "{{.Org}}"
+  name = "{{.IpSetName}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.nsxt-edge.id
+}
+`
+
+const testAccVcdNsxtIpSetInheritedVdcStep4 = `
+data "vcd_external_network_v2" "existing-extnet" {
+	name = "{{.ExternalNetwork}}"
+}
+
+resource "vcd_nsxt_edgegateway" "nsxt-edge" {
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
+  name = "{{.NsxtEdgeGatewayVcd}}"
+
+  external_network_id = data.vcd_external_network_v2.existing-extnet.id
+
+  subnet {
+     gateway               = tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].gateway
+     prefix_length         = tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].prefix_length
+
+     primary_ip            = tolist(tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].static_ip_pool)[0].end_address
+     allocated_ips {
+       start_address = tolist(tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].static_ip_pool)[0].end_address
+       end_address   = tolist(tolist(data.vcd_external_network_v2.existing-extnet.ip_scope)[0].static_ip_pool)[0].end_address
+     }
+  }
+}
+
+resource "vcd_nsxt_ip_set" "set1" {
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
```

Comment:
The test is using a field which has been deprecated in this PR?

Reply:
// * Step 1 - Rely on configuration coming from `provider` configuration for `vdc` value
// * Step 2 - Test that import works correctly
// * Step 3 - Test that data source works correctly
// * Step 4 - Start using `vdc` fields in resource and make sure it is not recreated
// * Step 5 - Test that import works correctly
// * Step 6 - Test data source

Yes, test validates that everything still works.